05_testing_and_ci/examples/python-testing/test_operations.py

```diff
@@ -0,0 +1,52 @@
+"""
+Tests for mathematical operations functions.
+"""
+
+from operations import find_max, find_average
+import pytest
+import csv
+
+
+# Unit test
+def test_find_max():
+    data = [43, 32, 167, 18, 1, 209]
+    expected_result = 209
+    actual_result = find_max(data)
+    assert actual_result == expected_result
+
+
+# Unit test
+def test_find_average():
+    data = [43, 32, 167, 18, 1, 209]
+    expected_result = 78.33
+    # expected_result = pytest.approx(78.3, abs=0.01)
+    actual_result = find_average(data)
+    assert actual_result == expected_result
+
+
+# Integration test
+def test_max_avg_compare():
+    """
+    Test to check that the average of a data set is always lesser than the maximum
+    """
+    data = [43, 32, 167, 18, 1, 209]
+    maximum = find_max(data)
+    average = find_average(data)
+    assert average < maximum
+
+
+# Regression test
+def test_average_reg():
```

Comment:
The name `reg` is vague. One could use `regression`, but I find this sounds ambiguous. 
```suggestion
def test_average_regression():
```

Maybe 

```suggestion
def regression_test_average():
```

would be more specific? However, in the latter case the functions names for the other tests are inconsistent as the function names do not specify the type of test is run. In this case one should maybe also rename the other function names.